web/themes/new_weather_theme/assets/js/components/combo-box.js

```diff
@@ -119,578 +120,687 @@ const comboTemplate = `
+    textInput.setAttribute("type", "hidden");
+    textInput.setAttribute("name", "placeName");
+    this.append(textInput);
+
+    if (formEl) {
+      const optionText = this.input.value;
+      textInput.value = optionText;
+
+      if (this.url) {
+        const result = {
+          text: this.input.value,
+          url: this.url,
+        };
+
+        this.saveSearchResult(result);
+        formEl.setAttribute("action", this.url);
+        return formEl.submit();
+      }
+
+      return this.getGeodataForKey(this.value).then((coordinates) => {
+        if (coordinates) {
+          const result = {
+            text: this.input.value,
+            url: `/point/${coordinates.lat}/${coordinates.lon}`,
+          };
+
+          this.saveSearchResult(result);
+
+          formEl.setAttribute("action", result.url);
+          formEl.submit();
         }
+      });
+    }
 
-        return Promise.reject(
-            new Error("No form ancestor element found for this combobox")
-        );
+    return Promise.reject(
+      new Error("No form ancestor element found for this combobox"),
+    );
+  }
+
+  /**
+   * Asynchronously fetches specific location data
+   * for a given search result, stashing it away in
+   * the cache (See ArcCache)
+   */
+  async cacheLocationGeodata(magicKey) {
+    if (!window.sessionStorage.getItem(magicKey)) {
+      const result = await getLocationGeodata(magicKey);
+      ArcCache.setItem(magicKey, result);
+    }
+  }
+
+  /**
+   * Attempts to retrieve location data for a search
+   * result by its magicKey from the cache.
+   * If not present, will make the Arc API call
+   * to fetch the data.
+   */
+  async getGeodataForKey(magicKey) {
+    const cached = ArcCache.getItem(magicKey);
+    if (!cached) {
+      return getLocationGeodata(magicKey);
     }
 
-    /**
-     * Asynchronously fetches specific location data
-     * for a given search result, stashing it away in
-     * the cache (See ArcCache)
-     */
-    async cacheLocationGeodata(magicKey){
-        if(!window.sessionStorage.getItem(magicKey)){
-            const result = await getLocationGeodata(magicKey);
-            ArcCache.setItem(magicKey, result);
-        }
+    return cached;
+  }
+
+  saveSearchResult(result) {
+    try {
+      // When loading previously-saved results, immediately remove any items for
+      // the same URL. We'll replace them with the new one. This way we don't
+      // end up having a whole list of the same place.
+      const saved = JSON.parse(
+        localStorage.getItem("wxgov_recent_locations") ?? "[]",
+      ).filter(({ url }) => url !== result.url);
+
+      // Put the new one at the front of the list.
+      saved.unshift(result);
+
+      localStorage.setItem(
+        "wxgov_recent_locations",
+        // We don't want to save everything forever. Just keep the most recent
+        // ten. We'll handle filtering down at display time.
+        JSON.stringify(saved.slice(0, 10)),
```

Comment:
Ok so recent 10 will show up in search selection, but only the recent 3 will show up if you haven't whittled the list down?

Reply:
The idea is to *keep* the most recent ten, but only show three. So if you don't type in anything, you'll get the three most recent. If you start typing, you'll get the three most recent *that match what you've typed*. Those two lists may not be the same. Eg, if your ten most recent locations are:

1. Portland
2. Pocatello
3. Philadelphia
4. Chicago
5. Portsmouth
6. Denver
7. Phoenix
8. Charlotte
9. Port Arthur
10. Minneapolis

Then when you first click the box, you'll get Portland, Pocatello, and Philadelphia. If you then start typing `Po`, your recent results will become Portland, Pocatello, and Portsmouth. If you back that out and start over with `C`, your recent results will be Chicago and Charlotte.

Does that make sense? Was this a bonkers decision on my part?

Reply:
I might ask product/design this question. It's totally fine by me and I quickly understood what was happening and it only happens on search which works well. I just wonder if it not being consistent is the issue. I understand because I read the code, but someone not reading the code might not?

Reply:
Paging @sarahmhgov on this for her thoughts.

I think this is good and something we can monitor in testing.